### ISSUE TITLE
Selecting multiple files and running command only runs it on one file

### ISSUE THREAD

**Issue by ADTC** [Alcyon-Dev/vscode-ext-git-skip#2]
As said on the tin ;) (It only runs on the file that I right-clicked on.)

PS: Thanks for this great extension. I often have the need to ignore certain changes, and this helps. I also learnt the difference between assume-unchanged and skip-worktree. Previously I was using assume-unchanged, but this was causing the file to be overwritten sometimes (for example in rebase). Hopefully skip-worktree will keep the changes during rebase but ignore them when committing.

**Comment by AlexDanault** [comment]
Good point, I'll look into it.

I'm glad the extension helps you, in return I'm thanking you for theses issues that will make the extension better !

**Comment by AlexDanault** [comment]
Humm, I can't reproduce this bug, if I shift-select multiple files, then right-click the last one and select skip/assume, all files get flagged.

Can you try it again, maybe record a video in case I can see what I am missing ?

**Comment by ADTC** [comment]
Video:


https://github.com/Alcyon-Dev/vscode-ext-git-skip/assets/6047296/af6d1b30-5543-4743-b1eb-daf8a5db401a

**Comment by ADTC** [comment]
When I run it on a selection of 6 files, 1 or sometimes 2 gets added. I noticed the following errors in the Extension Host output (2 passed, so I think this is 4 failures):

```
2024-04-04 14:39:06.431 [error] Error: command failed
	at ChildProcess.<anonymous> (/Users/adtc/.vscode/extensions/alcyon-dev.git-skip-0.2.1/node_modules/@npmcli/promise-spawn/lib/index.js:53:27)
	at ChildProcess.emit (node:events:514:28)
	at ChildProcess.emit (node:domain:489:12)
	at maybeClose (node:internal/child_process:1091:16)
	at Socket.<anonymous> (node:internal/child_process:449:11)
	at Socket.emit (node:events:514:28)
	at Socket.emit (node:domain:489:12)
	at Pipe.<anonymous> (node:net:323:12)
2024-04-04 14:39:06.433 [error] Error: command failed
	at ChildProcess.<anonymous> (/Users/adtc/.vscode/extensions/alcyon-dev.git-skip-0.2.1/node_modules/@npmcli/promise-spawn/lib/index.js:53:27)
	at ChildProcess.emit (node:events:514:28)
	at ChildProcess.emit (node:domain:489:12)
	at maybeClose (node:internal/child_process:1091:16)
	at Socket.<anonymous> (node:internal/child_process:449:11)
	at Socket.emit (node:events:514:28)
	at Socket.emit (node:domain:489:12)
	at Pipe.<anonymous> (node:net:323:12)
2024-04-04 14:39:06.434 [error] Error: command failed
	at ChildProcess.<anonymous> (/Users/adtc/.vscode/extensions/alcyon-dev.git-skip-0.2.1/node_modules/@npmcli/promise-spawn/lib/index.js:53:27)
	at ChildProcess.emit (node:events:514:28)
	at ChildProcess.emit (node:domain:489:12)
	at maybeClose (node:internal/child_process:1091:16)
	at Socket.<anonymous> (node:internal/child_process:449:11)
	at Socket.emit (node:events:514:28)
	at Socket.emit (node:domain:489:12)
	at Pipe.<anonymous> (node:net:323:12)
2024-04-04 14:39:06.434 [error] Error: command failed
	at ChildProcess.<anonymous> (/Users/adtc/.vscode/extensions/alcyon-dev.git-skip-0.2.1/node_modules/@npmcli/promise-spawn/lib/index.js:53:27)
	at ChildProcess.emit (node:events:514:28)
	at ChildProcess.emit (node:domain:489:12)
	at maybeClose (node:internal/child_process:1091:16)
	at Socket.<anonymous> (node:internal/child_process:449:11)
	at Socket.emit (node:events:514:28)
	at Socket.emit (node:domain:489:12)
	at Pipe.<anonymous> (node:net:323:12)
```

**Comment by AlexDanault** [comment]
Thanks for the reproduction video, I'm trying to see what is different between your setup and mine, but I don't see anything.

I'll add better error logging in the next version, I'll keep you up to date here once it is out.

**Comment by AlexDanault** [comment]
Version 0.3.0 is out now, with better error logging. If you could update and re-try this issue to get the full error messages, it'd be much appreciated.

Thanks !

**Comment by ADTC** [comment]
It now shows error notifications:

<img width="471" alt="image" src="https://github.com/Alcyon-Dev/vscode-ext-git-skip/assets/6047296/969297fa-4044-40fe-a34b-631153739531">

Seems like this could be fixed if you run just one command to add all the files, instead of concurrent commands. [See the documentation,](https://git-scm.com/docs/git-update-index) there it says `[<file>...]` so you can list more than one file on the same command line. Do test for any hard limit on the command line length though - you may need to batch (with wait intervals) if there are too many files.

<img width="687" alt="image" src="https://github.com/Alcyon-Dev/vscode-ext-git-skip/assets/6047296/b270cb7a-901f-47dc-9dd5-0b479fdfc5a1">


BTW: For safety, you may want to add `--` before listing the files.

**Comment by AlexDanault** [comment]
Should now be fixed in 0.4.1. If you can confirm this works for you, I'll close this issue.

**Comment by ADTC** [comment]
This works too. Thanks!
